configuration/klippy/beacon_mesh.py

```
@@ -33,142 +116,822 @@ def register_handler(self):
+		else:
+			return result
+
+	def _gaussian_filter(self, data, sigma, mode):
+		if not self.scipy_ndimage:
+			try:
+				self.scipy_ndimage = importlib.import_module("scipy.ndimage")
+			except ImportError:
+				raise Exception(
+					"Could not load `scipy.ndimage`. To install it, simply run `ratos doctor`. This "
+					"module is required for Beacon contact compensation mesh creation."
+				)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)